scio-core/src/main/scala/com/spotify/scio/runners/dataflow/DataflowResult.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.runners.dataflow
+
+import java.util.Collections
+
+import com.google.api.services.dataflow.model.{Job, JobMetrics}
+import com.spotify.scio.metrics.Metrics
+import com.spotify.scio.{RunnerResult, ScioResult}
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions
+import org.apache.beam.runners.dataflow.{DataflowClient, DataflowPipelineJob}
+import org.apache.beam.sdk.options.PipelineOptionsFactory
+import org.apache.beam.sdk.runners.AppliedPTransform
+import org.apache.beam.sdk.transforms.PTransform
+import org.apache.beam.sdk.values.{PInput, POutput}
+import org.apache.beam.sdk.{Pipeline, PipelineResult}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+/** Represent a Dataflow runner specific result. */
+class DataflowResult(val internal: DataflowPipelineJob) extends RunnerResult {
+
+  def this(internal: PipelineResult) = this(internal.asInstanceOf[DataflowPipelineJob])
+
+  private val client = DataflowResult.getOptions(internal.getProjectId).getDataflowClient
+
+  /** Get Dataflow [[Job]]. */
+  lazy val getJob: Job = client.projects().jobs()
```

Comment:
I think there should be retries here, a single failure of any of those calls would fail a whole workflow - usually at the very end.

Reply:
Actually `DataflowResult` can be re-constructed with just project and job ID, async, and independent of submission so maybe keep it simple here and let caller (some centralized service that monitors job completion) handle retry/fallback?